Add NusaX Bitext Mining (MINERS)

gentaiscool opened the pull request:
NusaX is a parallel dataset for machine translation and sentiment analysis on 11 Indonesian languages

## Checklist
<!-- Please do not delete this -->

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 


### Adding datasets checklist
<!-- see also https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md -->

**Reason for dataset addition**: ... <!-- Add reason for adding dataset here. E.g. it covers task/language/domain previously not covered -->

- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

KennethEnevoldsen commented on mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py:
```diff
@@ -0,0 +1,64 @@
+        eval_splits=["train"],
+        eval_langs=_LANGUAGES,
+        main_score="f1",
+        date=("2023-01-01", "2023-03-01"),
```
This is intended to be data creation dates (when it was written). Can be an estimate

gentaiscool replied:
I just updated the date range to `date=("2021-08-01", "2022-07-01")`

gentaiscool replied:
That was the time when we collected the dataset

KennethEnevoldsen commented on mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py:
```diff
@@ -0,0 +1,64 @@
+        }
+        """,
+        n_samples={"train": 500},
+        avg_character_length={},
```
required

gentaiscool replied:
I added the char length.

KennethEnevoldsen reviewed (commented):
Only a few minor comments. For points let us just use the point system for adding datasets. It is 2 points for the dataset + 4 pr. new language task pair. 2 points are also given for a review. We can either do it per. PR or at the end.

gentaiscool commented:
Thank you for the review. I have addressed your suggestion. @KennethEnevoldsen 

I found two issues:
1. when running `make test`
```
===================================================== short test summary info =====================================================
FAILED tests/test_RerankingEvaluator.py::TestRerankingEvaluator::test_nAUC - assert 0.9406528189910982 == 0.86943 ± 8.7e-05
```
It seemed unrelated to my change. I also got the error when running the command from the `main` branch.

2. And, I could not run `intfloat/multilingual-e5-small` model.
```
ValueError: Model None not found for model intfloat/multilingual-e5-small
```

gentaiscool commented:
I have completed the checklist.

I have addressed your comments @KennethEnevoldsen. I could not run `intfloat/multilingual-e5-base` since I got this error `ValueError: Model None not found for model intfloat/multilingual-e5-small` so as a replacement I evaluated the dataset on `sentence-transformers/LaBSE`.

KennethEnevoldsen commented on mteb/tasks/BitextMining/multilingual/NusaXBitextMining.py:
```diff
@@ -0,0 +1,64 @@
+        }
+        """,
+        n_samples={"train": 5500},
+        avg_character_length={"train": 5.00},
```
Is it on average 5 characters?

gentaiscool replied:
Sorry I put the number of words. Let me recalculate the number.

KennethEnevoldsen replied:
No worries. My guess was that it was something like that :)

KennethEnevoldsen commented:
> I could not run intfloat/multilingual-e5-base since I got this error ValueError: Model None not found for model intfloat/multilingual-e5-small so as a replacement I evaluated the dataset on sentence-transformers/LaBSE.

@gentaiscool will you add the results to the PR as well. 

> ValueError: Model None not found for model intfloat/multilingual-e5-small'

Addressed in #888 so merging from main should fix this

Will you add the points as well?

gentaiscool commented:
thank you @KennethEnevoldsen for the review and now the evaluation on e5 model works :). I have updated the metadata (avg num of characters -- previously I put the avg of char per word, now should be correct) and add the model results.

KennethEnevoldsen commented:
Points sounds reasonable. set tests to run and enable auto-merge if they pass
